File: jimmer-ddd-java-sample/domain/src/main/java/org/morecup/jimmerddd/java/sample/domain/order/OrderDetail.java
package org.morecup.jimmerddd.java.sample.domain.order;

import org.babyfish.jimmer.sql.Entity;
import org.babyfish.jimmer.sql.OneToOne;
import org.jetbrains.annotations.Nullable;
import org.morecup.jimmerddd.java.sample.domain.base.BaseEntity;

@Entity
public interface OrderDetail extends BaseEntity {
    
    String address();

    @OneToOne(mappedBy = "orderDetail")
    @Nullable
    Order order();
}

File: jimmer-ddd-java-sample/domain/src/main/java/org/morecup/jimmerddd/java/sample/domain/order/OrderDetail.kt
package org.morecup.jimmerddd.kotlin.sample.domain.order

import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.OneToOne
import org.morecup.jimmerddd.kotlin.sample.domain.base.BaseEntity


@Entity
interface OrderDetail: BaseEntity {

    val address: String

    @OneToOne(mappedBy = "orderDetail")
    val order: Order?
}